.github/workflows/update.yml

```diff
@@ -0,0 +1,50 @@
+name: update
+
+on:
+  workflow_dispatch:
+    inputs:
+      release:
+        description: 'Release (defaults to latest)'
+        required: false
+        default: 'source/master'
+
+jobs:
+  update:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up update
+        run: bash scripts/setup-update.sh ${{ github.event.inputs.release }}
```

Comment:
I separated the scripts into different files (`setup-update.sh`, `package-json-merge.js`, etc.) instead of making one giant shell script because I thought some of them may be useful. For example, I could imagine using `setup-update.sh` as part of some manual merge without wanting to push the results to GitHub.